src/smc-webapp/project_files.cjsx

```diff
@@ -1861,11 +1862,21 @@ ProjectFiles = (name) -> rclass
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
+    render_not_public_error : ->
+        if @props.redux.getStore('account').is_logged_in()
+            <ErrorDisplay title="Directory is not public" error={"You are trying to access a non public project that you are not a collaborator on. You need to ask a collaborator of the project to add you."} />
```

Comment:
Presumably we should make a new issue which is a mechanism for requesting to be added to a project, which would email the current collabs or something else (depending on their prefs). Google groups have something like this.